packages/f-checkout/src/components/Selector.vue

```diff
@@ -48,85 +27,38 @@ export default {
             return this.serviceType === CHECKOUT_METHOD_DELIVERY
                 ? this.$t('labels.deliveryOrderMethod')
                 : this.$t('labels.collectionOrderMethod');
-        }
-    },
+        },
 
-    watch: {
-        'fulfillment.times' (newValue) {
-            const selected = newValue.find(t => t.selected);
-            if (selected) {
-                this.selectedTime = selected.from;
-            }
+        fulfillmentTimes () {
+            /*
+            * Create an array from fulfillment times labels to
+            * display as options in dropdown
+            */
+            return this.fulfillment.times.map(time => time.label.text);
         }
     },
 
     methods: {
-        selectionChanged () {
-            this.fulfillment.times.forEach(el => { el.selected = false; });
-            const newSelected = this.fulfillment.times.find(t => t.from === this.selectedTime);
-            if (newSelected) {
-                newSelected.selected = true;
-            }
+        selectionChanged (e) {
+            /*
+            * Update all fulfillment.times.selected to false
+            * Update chosen fulfillment.times.selected to true
+            */
+            this.fulfillment.times.forEach(time => {
+                if (time.label.text === e) {
+                    time.selected = true;
+                } else {
+                    time.selected = false;
+                }
```

Comment:
Could simplify this to:
```suggestion
                time.selected = time.label.text === e;
```

Reply:
🤦‍♀️ Thanks